lib/spack/spack/test/llnl/util/filesystem.py

```diff
@@ -0,0 +1,214 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+"""Tests for ``llnl/util/filesystem.py``"""
+
+from spack.stage import Stage
+
+import llnl.util.filesystem as fs
+import os
+import pytest
+
+
+@pytest.fixture()
+def stage():
```

Comment:
I think the `pytest` way to do this is to use a [tmpdir_factory](https://docs.pytest.org/en/latest/tmpdir.html#the-tmpdir-factory-fixture):

```python
def stage(tmpdir_factory):
    s = tmpdir_factory.mktemp('filesystem_test')
    with s.as_cwd():
        # do stuff
    yield s
```

`s` is a `py` [`path`](https://py.readthedocs.io/en/latest/path.html#py-path) so instead of `stage.path` you have to do `str(stage)`.

You don't have to delete the directory, `pytest`'s fixtures handle it.  FWIW, you could also do `with s:` instead of using `create` and `destroy` directly -- we implemented that.  But, I've started using this and regular old `tmpdir` in tests, and I like it because the Spack tests aren't as inter-dependent.  Also if we refactor or replace the stage API, this doesn't have to change.